f/connectors/comapeo/comapeo_alerts.py

```diff
@@ -144,13 +156,56 @@ def filter_alerts(
         if alert.get("alert_id") not in alerts_posted_to_comapeo
     ]
 
+    logger.info(f"{len(unposted_alerts)} alerts in database not yet posted to CoMapeo.")
     return unposted_alerts
 
 
+def transform_alerts(alerts: list[dict]):
+    """
+    Transforms a list of alerts into a format that matches the expected schema on the CoMapeo API.
+
+    Parameters
+    ----------
+    alerts : list[dict]
+        A list of dictionaries, where each dictionary represents an alert.
+
+    Returns
+    -------
+    list[dict]
+        A list of dictionaries, where each dictionary represents an alert in a format that can be posted to the CoMapeo API.
+    """
+    logger.info("Transforming alerts...")
+
+    transformed_alerts = [
+        {
+            # CoMapeo API requires these to be ISO 8601 format
+            "detectionDateStart": datetime.strptime(
+                alert["date_start_t0"], "%Y-%m-%d"
+            ).isoformat(timespec="seconds")
+            + "Z",
```

Comment:
`"%Y-%m-%d"` is already a valid ISO-8601 _date_, but maybe CoMapeo's requirement is that the value be a _datetime_.


Just want to double-check that the addition of time here is happening as you expect:
- is midnight an acceptable time? In particular for `detectionDateEnded`, if the end was, say noon on that date, then your new timestamp would be before the actual end.
- is UTC correct? Do we have any way to infer a more correct timezone?


Just ideas -- I trust you've thought about this already.

Reply:
Thanks, noted the datetime requirement in the comment.

Yeah, this is quite tricky. There is no deterministic way to infer. But your notes made me want to adapt the timestamp to 15:00 UTC which is an expected time for when the alerts provider might post alerts on the day.